crates/bevy_ecs/src/schedule/executor_parallel.rs

```diff
@@ -212,11 +215,39 @@ impl ParallelExecutor {
             }
             // Queue the system if it has no dependencies, otherwise reset its dependency counter.
             if system_data.dependencies_total == 0 {
-                self.queued.insert(index);
+                // Non-send systems are incompatible, as they must all be executed on the main thread.
+                if should_run
+                    && (!self.non_send_running || system_data.is_send)
```

Comment:
I think we should reuse the `can_run` function here. This is critical code for ensuring that conflicting systems don't run at the same time and if that code gets changed it should be only changed in one place.

I did this here https://github.com/bevyengine/bevy/compare/main...hymm:prepare_try_start if you want to copy.

Reply:
We cannot. That requires a &self reference which does not work with the &mut held above.

Reply:
did you look at the code I linked?

Reply:
Nope, apparently I didn't. Adapted it for this PR.